### PR TITLE
FIX-#2844: Check for both types of ray.ObjectRef

### LINKED ISSUE #2844: Fix defensive assertions for Ray Client users
Related to https://github.com/ray-project/ray/issues/14042

For now, we will check for both types.

### PR DESCRIPTION
**Pull request by devin-petersohn**
Resolves #2844

Signed-off-by: Devin Petersohn <devin.petersohn@gmail.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2844 
- [ ] tests added and passing

### CODE REVIEW

File: modin/engines/ray/pandas_on_ray/frame/partition.py [hunk 1]
```diff
@@ -20,11 +20,18 @@
 import ray
 from ray.worker import RayTaskError
 from ray.services import get_node_ip_address
+from packaging import version
+ObjectIDType = ray.ObjectRef
```
Comment (RehanSD) [inline modin/engines/ray/pandas_on_ray/frame/partition.py:24]: Is this a good name @devin-petersohn, or would you prefer something like `object_id_cls`?

File: modin/engines/ray/pandas_on_ray/frame/partition.py [hunk 2]
```diff
@@ -206,7 +213,7 @@ def ip(self):
                 self.drain_call_queue()
             else:
                 self._ip_cache = self.apply(lambda df: df)._ip_cache
-        if isinstance(self._ip_cache, ray.ObjectID):
+        if isinstance(self._ip_cache, (ray.ObjectID, ClientObjectRef)):
```
Comment (RehanSD) [inline modin/engines/ray/pandas_on_ray/frame/partition.py:216]:
```suggestion
        if isinstance(self._ip_cache, ObjectIDType):
```

**Review by williamma12 — COMMENTED** [review]
This fix by itself looks good to me. 

However, would it be cleaner if we made `ObjectIDType` a class variable instead? Every engine would have to know its own `ObjectIDType` so that can add some more consistency between the different children of `BaseFramePartition`

**Comment by richardliaw** [comment]
hey @devin-petersohn is there anything blocking this PR?

**Comment by devin-petersohn** [comment]
@richardliaw Sorry for the delay. I don't think there are any specific blockers here.

@williamma12 I think it would be fine to merge this and add that minor refactoring as a future step. This is still consistent and correct at least. We will need to add some testing for Ray client, which we don't have set up yet.

**Review by williamma12 — APPROVED** [review]
Sounds good! Let's plan to refactor this soon

**Pull request merged**